multiqc/modules/verifybamid/verifybamid.py

```diff
@@ -0,0 +1,238 @@
+                for s_name in parsed_data:
+                    # add the sample as a key to the verifybamid_data dictionary and the dictionary of values as the value
+                    self.verifybamid_data[s_name] = parsed_data[s_name]
+        
+        # Filter to strip out ignored sample names
+        self.verifybamid_data = self.ignore_samples(self.verifybamid_data)
+
+        if len(self.verifybamid_data) == 0:
+            raise UserWarning
+
+        # print number of verifyBAMID reports found and parsed
+        log.info("Found {} reports".format(len(self.verifybamid_data)))
+
+        # Write parsed report data to a file	
+        self.write_data_file(self.verifybamid_data, 'multiqc_verifybamid')
+
+        # add to General Stats Table
+        self.verifybamid_general_stats_table()
+
+        # add section with the values from the verify BAM ID output
+        self.verifybamid_table()
+
+
+    def parse_selfsm(self, f):
+        """ Go through selfSM file and create a dictionary with the sample name as a key, """
+        #create a dictionary to populate from this sample's file
+        parsed_data = dict()
+        # set a empty variable which denotes if the headers have been read
+        headers = None
+        # for each line in the file
+        for l in f['f'].splitlines():
+                # split the line on tab
+                s = l.split("\t")
+                # if we haven't already read the header line
+                if headers is None:
+                    # remove the leading hash from the first element of the header
+                    s[0] = s[0].lstrip('#')
+                    # assign this list to headers variable
+                    headers = s
+                # for all rows after the first
+                else:
+                    s_name=s[0]
+                    s_name = self.clean_s_name(s_name, f['root'])
+                    # create a dictionary entry with the first column as a key (sample name) and an empty dictionary as a value
+                    parsed_data[s_name] = {}
+                    # for each item in list of items in the row
+                    for i, v in enumerate(s):
+                    # if it's not the first element (if it's not the name)
+                        if i != 0:
+                            # try and convert the value into a float
+                            try:
+                            # and add to the dictionary the key as the corrsponding item from the header and the value from the list
+                                parsed_data[s_name][headers[i]] = float(v)
+                                #if can't convert to float...
+                            except ValueError:
+                            # add to the dictionary the key as the corrsponding item from the header and the value from the list
+                                parsed_data[s_name][headers[i]] = v
+    
+        # nothing has been parsed return None
+        if len(parsed_data) == 0:
```

Comment:
Again, pedantic, but as you proceed to loop through `parsed_data` once it's returned, you don't really need this check. If `len(parsed_data)` is `0` then the proceeding looping code won't run and you'll get the same effect with less code.